src/Compilers/CSharp/Test/Semantic/Semantics/ForEachTests.cs

```diff
@@ -3196,5 +3196,137 @@ static void Main()
         { }").WithArguments("System.ValueType", "MissingBaseType1, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null").WithLocation(6, 9)
                 );
         }
+
+        [Fact]
+        [WorkItem(26585, "https://github.com/dotnet/roslyn/issues/26585")]
+        public void ForEachIteratorWithCurrentRefKind_Async_Ref()
+        {
+            CreateCompilation(@"
+using System.Threading.Tasks;
+
+class E
+{
+    public class Enumerator
+    {
+        public ref int Current => throw null;
+        public bool MoveNext() => throw null;
+    }
+
+    public Enumerator GetEnumerator() => new Enumerator();
+}
+class C
+{
+    public async static Task Test()
+    {
+        await Task.CompletedTask;
+
+        foreach (ref int x in new E())
+        {
+            System.Console.Write(x);
+        }
+    }
+}").VerifyDiagnostics(
+                // (20,26): error CS8177: Async methods cannot have by reference locals
```

Comment:
>by reference [](start = 68, length = 12)

Nit: consider "by-reference" as we do in [CS8172](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDZICYgNQA+AAgAwAERAjANwCwAUKRZQKx2MDMFATGQMJkA3gzKiKXIgA4KANjIBZABQBKISLEa0AOxhkAnmQC8ZEuw0aEcAGZltugB5H9ZsQF8GroA)